### ISSUE TITLE
834 (Benefit Enrollment and Maintenance)

### ISSUE THREAD

**Issue by cole-maxwell1** [apimeister/x12-types#1]
Hi,

I want to say that this is a neat collection of packages, and I would certainly be interested in collaborating on this. However, I am still quite new to both EDI and Rust so I may need some additional guidance. I work in the dental insurance industry in the US. I am interested in bringing rust into our organization. We have some heavy EDI parsing tasks where I think Rust in combination with `serde` could really shine.

I'd like to collaborate on developing a type for the [834 Benefit Enrollment and Maintenance spec](https://www.dhs.wisconsin.gov/publications/p00264.pdf). If you would also be interested, I would enjoy getting my hands dirty on some rust in this project. I can also try battle-testing serialization and deserialization on some of my real data at work to provide feedback.

Would you be interested in also putting together a contribution guide for the project? It would be helpful for me to know where I should get started and how I can help.

**Comment by JensWalter** [comment]
I put together some very basic support for 834. I recently shifted the parser implementation to `nom` because the previous approach did not work with more complex EDIs. Sadly I still have no idea how stable the current implementation is. So all of this is still a learning-by-doing (so no contribution guide whatsoever).
I think most errors will appear when this library is used with real data. I also added some test cases, but this only covers a very minimal feature set.
[https://github.com/apimeister/x12-types/blob/main/src/v005010/test_834.rs](https://github.com/apimeister/x12-types/blob/main/src/v005010/test_834.rs)
